### PR TITLE
Added suport for Talon files

### PR DESCRIPTION
**Pull request by Nova38**
## Description

This pull request is adding support for Talon language. It is a scripting language that is used by [Talon](https://talonvoice.com/) to define and map voice commands to actions it can take. Talon is a hands-free input system for Windows, MacOS, and Linux. 
## Checklist:

- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Atalon
  - [X] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/knausj85/knausj_talon/blob/14a370f18f248ba19177a2543c49335b415a1464/text/line_commands.talon
      - https://github.com/knausj85/knausj_talon/blob/master/text/generic_editor.talon
      - https://github.com/Nova38/knausj_talon/master/text/find_and_replace.talon
    - Sample license(s):
      - All are distributed under a MIT license
  
  - [X] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Fmrob95%2Fvscode-TalonScript%2Fblob%2Fmaster%2Fsyntaxes%2Ftalon.tmLanguage.yml&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fmrob95%2Fvscode-TalonScript%2Fblob%2Fmaster%2Fsyntaxes%2Ftalon.tmLanguage.yml&code=%23+activate+this+.talon+file+if+the+current+app+name+is+%22Chrome%22%0D%0A%23+you+can+find+app+names+by+running+ui.apps%28%29+in+the+REPL%0D%0Aapp.name%3A+Chrome%0D%0A-%0D%0A%23+key_wait+increases+the+delay+when+pressing+keys+%28milliseconds%29%0D%0A%23+this+is+useful+if+an+app+seems+to+jumble+or+drop+keys%0D%0Asettings%28%29%3A%0D%0A++++key_wait+%3D+4.0%0D%0A%0D%0A%23+activate+the+global+tag+%22browser%22%0D%0Atag%28%29%3A+browser%0D%0A%0D%0A%23+define+some+voice+commands%0D%0Ahello+chrome%3A+%22hello+world%22%0D%0Aswitch+tab%3A+key%28ctrl-tab%29%0D%0Ago+to+google%3A%0D%0A++++key%28ctrl-t%29+%23+note%3A+use+cmd-t+on+Mac%0D%0A++++insert%28%22google.com%22%29%0D%0A++++key%28enter%29
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by Nova38** [comment]
Just wanted to check in and see if there is anything else i need to add to this pull request?

Thanks

**Review by lildude — CHANGES_REQUESTED** [review]

File: samples/Talon/example.talon [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# activate the global tag "browser"
+tag(): browser
+
+# define some voice commands
+hello chrome: "hello world"
+switch tab: key(ctrl-tab)
+go to google:
+    # note: use key(cmd-t) on Mac
+    key(ctrl-t)    insert("google.com")
+    key(enter)
```
[the remark this answers is not in the document]
Comment (Nova38) [reply samples/Talon/example.talon:19]: Okay, sorry about that. How many samples is ideal to include? (The line_commands file included in the pull request is one real world example, I have others real world examples I can add, but don't won't to overdue it)

[the remark this answers is not in the document]
Comment (lildude) [reply samples/Talon/example.talon]: One or two should be fine. Two would be best as it'll add a bit more data for the classifier to work with in the event another language is added with the same extension.

[the remark this answers is not in the document]
Comment (Nova38) [reply samples/Talon/example.talon]: Sounds good. I went ahead and added two more samples to the pull request and added the links to the initial pull checklist. I as well removed the "hello world" samples info from the checklist.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -6214,6 +6214,14 @@ TXL:
   tm_scope: source.txl
   ace_mode: text
   language_id: 366
+Talon:
+  type: programming
+  ace_mode: text
+  color: "#333333"
```
Comment (lildude) [inline lib/linguist/languages.yml:6220]: Do you really want this grey colour `#333333`?

**Review by lildude — CHANGES_REQUESTED** [review]
I've removed the HTML file you accidentally added and added a comment about the colour.

Please can you address the merge conflicts too.

**Review by Nova38 — COMMENTED** [review]
I Believe everything should be in-order now

**Pull request merged**